packages/api/src/server/application/service/application.factory.ts

```diff
@@ -666,15 +683,16 @@ export class ApplicationFactory {
   generateApplicationRequestFromGit(checkGitRegistry: Application[], tmp: Environment, gitRequestDTO: GitDeploymentRequestDTO): CreateApplicationDto {
     const applicationRequest = new CreateApplicationDto();
     const existingDeployment = checkGitRegistry.find((i) => i.env === tmp.env);
-    applicationRequest.name = existingDeployment?.name || checkGitRegistry[0].name;
```

Comment:
Line time when we've discussed, we came to this conclusion  that Nest handles it so we won't use this, else we're not 

`How are we handling the scenario when checkGitRegistry.find returns undefined?` : We won't be undefined, because before calling this function we're checking that do we have value for the `checkGitRegistry` or not, if we don't have atlease 1 element, we won't call this function.

Reply:
Sure I'll add an exception to handle it
`if(!checkGitRegistry) this.exceptionService.badRequestException({ message: "Application is not registered" })`